core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/common/AbstractUserStoreManager.java

```diff
@@ -1736,22 +1742,28 @@ public final void addUser(String userName, Object credential, String[] roleList,
                 }
             }
 
-            if (claims != null) {
-                for (Map.Entry<String, String> entry : claims.entrySet()) {
-                    ClaimMapping claimMapping = null;
-                    try {
-                        claimMapping = (ClaimMapping) claimManager.getClaimMapping(entry.getKey());
-                    } catch (org.wso2.carbon.user.api.UserStoreException e) {
-                        String errorMessage = "Error in obtaining claim mapping for persisting user attributes.";
-                        throw new UserStoreException(errorMessage, e);
-                    }
-                    if (claimMapping == null) {
-                        String errorMessage = "Invalid claim uri has been provided: " + entry.getKey();
-                        throw new UserStoreException(errorMessage);
-                    }
+            for (Map.Entry<String, String> entry : claims.entrySet()) {
+                ClaimMapping claimMapping = null;
+                try {
+                    claimMapping = (ClaimMapping) claimManager.getClaimMapping(entry.getKey());
+                } catch (org.wso2.carbon.user.api.UserStoreException e) {
+                    String errorMessage = "Error in obtaining claim mapping for persisting user attributes.";
+                    throw new UserStoreException(errorMessage, e);
+                }
+                if (claimMapping == null) {
+                    String errorMessage = "Invalid claim uri has been provided: " + entry.getKey();
+                    throw new UserStoreException(errorMessage);
                 }
             }
 
+            User user = new DefaultUser();
+            user.setUsername(userName);
+
+            // TODO: We might need to pick the id manager from a config.
+            // TODO: We need a method to get the correct data source. Think about a Id manger factory.
+            IdManager idManager = new JDBCUserIdManager(null);
```

Comment:
Can we add a setter for IdManager, initialized to default at the class constructor.